assets/src/edit-story/components/panels/colorPreset/index.js

```diff
@@ -0,0 +1,283 @@
+  flex-wrap: wrap;
+  overflow: auto;
+`;
+
+const ButtonWrapper = styled.div`
+  flex-basis: 16%;
+  height: 35px;
+`;
+
+function ColorPresetPanel() {
+  const {
+    state: {
+      selectedElementIds,
+      selectedElements,
+      story: { colorPresets },
+    },
+    actions: { updateStory, updateElementsById },
+  } = useStory();
+
+  const [isEditMode, setIsEditMode] = useState(false);
+
+  const ref = useRef();
+
+  const {
+    actions: { showColorPickerAt, hideSidebar },
+  } = useSidebar();
+
+  const openColorPicker = useCallback(
+    (evt) => {
+      evt.stopPropagation();
+      showColorPickerAt(ref.current, {
+        onChange: () => null,
+        hasGradient: true,
+        hasOpacity: true,
+        onClose: hideSidebar,
+        addActions: getColorPickerActions,
+      });
+    },
+    [showColorPickerAt, hideSidebar]
+  );
+
+  const handleDeleteColor = useCallback(
+    (toDelete) => {
+      const colors = colorPresets.filter((color) => color !== toDelete);
+      updateStory({
+        properties: {
+          colorPresets: colors,
+        },
+      });
+    },
+    [colorPresets, updateStory]
+  );
+
+  const handleApplyBackgroundColor = useCallback(
+    (color) => {
+      updateElementsById({
+        elementIds: selectedElementIds,
+        properties: (currentProperties) => {
+          const { type } = currentProperties;
+          // @todo Is this necessary?
+          const { isMedia } = getDefinitionForType(type);
+          return isMedia
+            ? {}
+            : {
+                backgroundColor: color,
+              };
+        },
+      });
+    },
+    [selectedElementIds, updateElementsById]
+  );
+
+  const isText =
+    selectedElements.length > 0 &&
+    selectedElements.every(({ type }) => 'text' === type);
+
+  const handleApplyTextColor = useCallback(
+    (color) => () => {
+      if (isText) {
+        updateElementsById({
+          elementIds: selectedElementIds,
+          properties: { color },
+        });
+      }
+    },
+    [isText, selectedElementIds, updateElementsById]
+  );
+
+  const getSecondaryActions = () => {
+    return !isEditMode ? (
+      <>
+        <EditModeButton
+          onClick={(evt) => {
+            evt.stopPropagation();
+            setIsEditMode(true);
+          }}
+        >
+          <Edit />
+        </EditModeButton>
+        <AddColorPresetButton ref={ref} onClick={openColorPicker}>
+          <Add />
+        </AddColorPresetButton>
+      </>
+    ) : (
+      <ExitEditMode
+        onClick={(evt) => {
+          evt.stopPropagation();
+          setIsEditMode(false);
+        }}
+      >
+        {__('Exit', 'web-stories')}
+      </ExitEditMode>
+    );
+  };
+
+  return (
+    <Panel name="colorpreset">
+      <PanelTitle
+        isPrimary
+        secondaryAction={getSecondaryActions()}
+        canCollapse={!isEditMode}
+      >
+        {__('Color presets', 'web-stories')}
+      </PanelTitle>
+      {colorPresets && (
```

Comment:
opacity should default to 100 not 0. It is really weird this way around.

Reply:
Yep, there might be an issue for it somewhere, this happens everywhere currently and is quite annoying.

Reply:
#673?